gtsam/slam/ReadMe.md

```diff
@@ -0,0 +1,64 @@
+# SLAM Factors
+
+## GenericProjectionFactor (defined in ProjectionFactor.h)
+
+Non-linear factor for a constraint derived from a 2D measurement. 
+The calibration is assumed known and passed in the constructor.
+The main building block for visual SLAM.
+
+Templated on
+- `POSE`, default `Pose3`
+- `LANDMARK`, default `Point3`
+- `CALIBRATION`, default `Cal3_S2`
+
+## SmartFactors
+
+These are "structure-less" factors, i.e., rather than introducing a new variable for an observed 3D point or landmark, a single factor is created that provides a multi-view constraint on several poses and/or cameras.
+
+### SmartFactorBase
+
+This is the base class for smart factors, templated on a `CAMERA` type.
+It has no internal point, but it saves the measurements, keeps a noise model, and an optional sensor pose.
+
+### SmartProjectionFactor
+
+Also templated on `CAMERA`. Triangulates a 3D point and keeps an estimate of it around.
+This factor operates with monocular cameras, and is used to optimize the camera pose
+*and* calibration for each camera, and requires variables of type `CAMERA` in values.
+
+If the calibration is fixed use `SmartProjectionPoseFactor` instead!
+
+
+### SmartProjectionPoseFactor
+
+Derives from `SmartProjectionFactor` but is templated on a `CALIBRATION` type, setting `CAMERA = PinholePose<CALIBRATION>`.
+This factor assumes that the camera calibration is fixed and the same for all cameras involved in this factor.
+The factor only constrains poses.
+
+If the calibration should be optimized, as well, use `SmartProjectionFactor` instead!
+
+### SmartProjectionFactorP
```

Comment:
I picked this name, but I'm not in love with it. Maybe we should rename using a different keyword? (multiCam?) Not sure if you really want to stick to the convention using the suffix "P", "PP" to indicate which variables are optimized.

Reply:
I think we need to have a design conversation, in general, to agree on "the right way" to handle multi-camera rigs.

Reply:
happy to!

Reply:
I'll give it a think, now that it's swapped into my brain, then ping you for a chat :-)

Reply:
ok thanks!